src/interval.jl

```diff
@@ -187,6 +187,72 @@ Base.isopen(interval::AbstractInterval{T,L,R}) where {T,L,R} = L === Open && R =
 isunbounded(interval::AbstractInterval{T,L,R}) where {T,L,R} = L === Unbounded && R === Unbounded
 isbounded(interval::AbstractInterval{T,L,R}) where {T,L,R} = L !== Unbounded && R !== Unbounded
 
+# Minimum and maximum calls for Closed and Unbounded Intervals, just return first and last.
+function Base.minimum(interval::AbstractInterval{T,L,R}; increment=nothing) where {T,L,R}
+    first(interval) === nothing && return typemin(T)
+    return first(interval)
+end
+
+function Base.maximum(interval::AbstractInterval{T,L,R}; increment=nothing) where {T,L,R}
+    last(interval) === nothing && return typemax(T)
+    return last(interval)
+end
+
+# Minimum and maximum calls for open bounds and ambigous types, return open side +- increment
+# Default increment val is eps(T)
+function Base.minimum(interval::AbstractInterval{T,Open,R}; increment=eps(T)) where {T,R}
+    isempty(interval) && return nothing
+    min_val = first(interval)
+    !isfinite(min_val) && return typemin(T)
+    min_val + increment ∈ interval && return min_val + increment
+    throw(DomainError(interval, "The increment is too large, $(min_val + increment) is not in the interval"))
+end
+
+function Base.maximum(interval::AbstractInterval{T,L,Open}; increment=eps(T)) where {T,L}
+    isempty(interval) && return nothing
+    max_val = last(interval)
+    !isfinite(max_val) && return typemax(T)
+    max_val - increment ∈ interval && return max_val - increment
+    throw(DomainError(interval, "The increment is too large, $(max_val - increment) is not in the interval"))
+end
+
+# Minimum and maximum calls for open bounds and Integer types, return open side +- increment
+# Default increment val for Integers is one(T)
+function Base.minimum(interval::AbstractInterval{T,Open,R}) where {T<:Integer,R}
+    return minimum(interval, increment=one(T))
+end
+
+function Base.maximum(interval::AbstractInterval{T,L,Open}) where {T<:Integer,L}
+    return maximum(interval, increment=one(T))
+end
+
+# Minimum and maximum calls for open bounds and Float types, return open side +- increment
+# Default increment val for Floats is nothing, which returns nextfloat or prevfloat (i.e., +- eps())
+function Base.minimum(interval::AbstractInterval{T,Open,R}; increment=nothing) where {T<:AbstractFloat,R}
+    isempty(interval) && return nothing
+    min_val = first(interval)
+    # Since intervals can't have NaN, we can just use !isfinite to check if infinite
+    next_val = if !isfinite(min_val) || increment === nothing || increment == eps(T)
```

Comment:
I'd only use `nextfloat` if a increment isn't explicitly given. Even if it is `eps(T)`
```suggestion
    next_val = if !isfinite(min_val) || increment === nothing
```